RIOT-OS/RIOT
# pkg/nimble: version bump to NimBLE 1.5.0 RC1

### Contribution description
NimBLE is about to be released in version 1.5.0. In order to update the NimBLE version in RIOT, too, there are some changes needed to the NimBLE port to cater for internal changes in NimBLE. These are most notably the use of Nordic `nrfx` headers in the NimBLE radio drivers and an overhaul of the BLE transport implementation in NimBLE.

This PR contains all the needed changes to RIOT to allow for updating NimBLE to its newest version:
- it adds the official `nrfx` headers as optional package to RIOT
- adapts the NimBLE Makefiles to cater for the BLE transport changes
- and of course updates NimBLE to its current version

### Testing procedure
- run and verify the NimBLE test applications from the examples folder
- run some IP over BLE tests using `gnrc_networking` on some NRF hardware

### Issues/PRs references
Depends on https://github.com/apache/mynewt-nimble/pull/1249
